stan/math/prim/meta/is_floating_point.hpp

```diff
@@ -0,0 +1,29 @@
+#ifndef STAN_MATH_PRIM_META_IS_FLOATING_POINT_HPP
+#define STAN_MATH_PRIM_META_IS_FLOATING_POINT_HPP
+
+#include <type_traits>
+
+namespace stan {
+
+/**
+ * Checks if decayed type is a floating point type
+ * @tparam The type to check
+ * @ingroup type_trait
+ */
+template <typename T>
+using is_floating_point = std::is_floating_point<std::decay_t<T>>;
+
+template <typename T>
+inline constexpr bool is_floating_point_v = is_floating_point<T>::value;
```

Comment:
just so the compiler does not get confused about which
```suggestion
inline constexpr bool is_floating_point_v = stan::is_floating_point<T>::value;
```

Reply:
This goes for all the other places as well